src/backend/base/langflow/base/models/aiml_constants.py

```diff
@@ -1,252 +1,46 @@
-    "stable-diffusion-v3-medium",
-    "teknium/OpenHermes-2-Mistral-7B",
-    "teknium/OpenHermes-2p5-Mistral-7B",
-    "togethercomputer/CodeLlama-13b-Instruct",
-    "togethercomputer/CodeLlama-13b-Python",
-    "togethercomputer/CodeLlama-34b",
-    "togethercomputer/CodeLlama-34b-Instruct",
-    "togethercomputer/CodeLlama-34b-Python",
-    "togethercomputer/CodeLlama-7b-Instruct",
-    "togethercomputer/CodeLlama-7b-Python",
-    "togethercomputer/Koala-13B",
-    "togethercomputer/Koala-7B",
-    "togethercomputer/LLaMA-2-7B-32K",
-    "togethercomputer/Llama-2-7B-32K-Instruct",
-    "togethercomputer/Llama-3-8b-chat-hf-int4",
-    "togethercomputer/Llama-3-8b-chat-hf-int8",
-    "togethercomputer/SOLAR-10.7B-Instruct-v1.0-int4",
-    "togethercomputer/StripedHyena-Hessian-7B",
-    "togethercomputer/StripedHyena-Nous-7B",
-    "togethercomputer/alpaca-7b",
-    "togethercomputer/evo-1-131k-base",
-    "togethercomputer/evo-1-8k-base",
-    "togethercomputer/guanaco-13b",
-    "togethercomputer/guanaco-33b",
-    "togethercomputer/guanaco-65b",
-    "togethercomputer/guanaco-7b",
-    "togethercomputer/llama-2-13b",
-    "togethercomputer/llama-2-13b-chat",
-    "togethercomputer/llama-2-70b",
-    "togethercomputer/llama-2-70b-chat",
-    "togethercomputer/llama-2-7b",
-    "togethercomputer/llama-2-7b-chat",
-    "togethercomputer/m2-bert-80M-2k-retrieval",
-    "togethercomputer/m2-bert-80M-32k-retrieval",
-    "togethercomputer/m2-bert-80M-8k-retrieval",
-    "upstage/SOLAR-10.7B-Instruct-v1.0",
-    "voyage-2",
-    "voyage-code-2",
-    "voyage-finance-2",
-    "voyage-large-2",
-    "voyage-large-2-instruct",
-    "voyage-law-2",
-    "voyage-multilingual-2",
-    "wavymulder/Analog-Diffusion",
-    "zero-one-ai/Yi-34B",
-    "zero-one-ai/Yi-34B-Chat",
-    "zero-one-ai/Yi-6B",
-]
+from typing import Any
 
-AIML_EMBEDDING_MODELS = [
-    "text-embedding-3-large",
-    "text-embedding-3-small",
-    "text-embedding-ada-002",
-    "text-multilingual-embedding-002",
-    "textembedding-gecko-multilingual@001",
-    "textembedding-gecko@001",
-    "textembedding-gecko@003",
-]
+import httpx
+
+
+class AimlModels:
+    def __init__(self):
+        self.chat_models = []
+        self.image_models = []
+        self.embedding_models = []
+        self.stt_models = []
+        self.tts_models = []
+        self.language_models = []
+
+    def get_aiml_models(self) -> list[dict[str, Any]]:
+        try:
+            with httpx.Client() as client:
+                response = client.get("https://api.aimlapi.com/models")
+                response.raise_for_status()
+                models = response.json().get("data", [])
+                self.separate_models_by_type(models)
+        except httpx.RequestError as e:
+            self.get_exception_message(e)
+        except httpx.HTTPStatusError as e:
+            self.get_exception_message(e)
+        except ValueError as e:
+            self.get_exception_message(e)
```

Comment:
I think we'd want to return an `err` here

Reply:
Yes, we did this, but within the function:

```python
def get_exception_message(e: Exception):
    return str(e)
```

I implemented this to match the code in `models.py`, which you can view in the file path `src/backend/base/langflow/base/models/model.py`, lines 42-43.
```python
def _get_exception_message(self, e: Exception):
    return str(e)
```

Or u mean that we want to raise error?

Reply:
Ah, yeah that's probably a misleading method we should get rid of. Yes, we should raise errors instead. 

If you wanted to go a little further, you can separate these like: 
```
try:
    with httpx.Client() as client:
        try:
            response = client.get("https://api.aimlapi.com/models")
            response.raise_for_status()
        except httpx.RequestError as e:
            raise APIConnectionError("Failed to connect to the API.") from e
        except httpx.HTTPStatusError as e:
            raise APIResponseError(f"API responded with error: {e.response.status_code}") from e

    try:
        models = response.json().get("data", [])
        self.separate_models_by_type(models)
    except <something> as e:
        raise ValueError("Failed to ....") from e
```